web/src/app/services/job/job.service.ts

```diff
@@ -70,6 +73,24 @@ export class JobService {
     );
   }
 
+  /**
+   * Returns a new job which is an exact copy of the specified job, but with new UUIDs for all items recursively.
+   */
+  createDuplicatedJob(job: Job, jobs: Map<string, Job> | undefined): Job {
```

Comment:
passing `job` and `jobs` seems redundant - maybe just pass `Survey` and the id of the job to duplicate?

Reply:
I've solved passing the color as already done with createNewJob.